cirq/contrib/svg/svg.py

```diff
@@ -0,0 +1,139 @@
+
+    for i in range(len(col_widths)):
+        # horizontal_padding seems to only zero-out certain paddings
+        # we use col_padding as a default
+        padding = tdd.horizontal_padding.get(i, col_padding)
+        col_widths[i] += padding
+
+    col_starts = [0.0]
+    for i in range(1, max_xi + 3):
+        col_starts.append(col_starts[i - 1] + col_widths[i - 1])
+
+    return col_starts, col_widths
+
+
+def tdd_to_svg(
+        tdd: 'cirq.TextDiagramDrawer',
+        ref_rowheight: float = 60,
+        ref_boxwidth: float = 40,
+        ref_boxheight: float = 40,
+        col_padding: float = 20,
+        y_top_pad: float = 5,
+) -> str:
+    height = tdd.height() * ref_rowheight
+    col_starts, col_widths = _fit_horizontal(tdd=tdd,
+                                             ref_boxwidth=ref_boxwidth,
+                                             col_padding=col_padding)
+
+    t = f'<svg width="{col_starts[-1]}" height="{height}">'
+
+    for yi, xi1, xi2, _ in tdd.horizontal_lines:
+        xi1 = cast(int, xi1)
+        xi2 = cast(int, xi2)
+        y = yi * ref_rowheight + y_top_pad + ref_boxheight / 2
+        x1 = col_starts[xi1] + col_widths[xi1] / 2
+        x2 = col_starts[xi2] + col_widths[xi2] / 2
+        t += f'<line x1="{x1}" x2="{x2}" y1="{y}" y2="{y}" ' \
+             f'stroke="black" stroke-width="1" />'
+
+    for xi, yi1, yi2, _ in tdd.vertical_lines:
+        y1 = yi1 * ref_rowheight + y_top_pad + ref_boxheight / 2
+        y2 = yi2 * ref_rowheight + y_top_pad + ref_boxheight / 2
+
+        xi = cast(int, xi)
+        x = col_starts[xi] + col_widths[xi] / 2
+        t += f'<line x1="{x}" x2="{x}" y1="{y1}" y2="{y2}" ' \
+             f'stroke="black" stroke-width="3" />'
+
+    for (xi, yi), v in tdd.entries.items():
+        x = col_starts[xi] + col_widths[xi] / 2
+        y = yi * ref_rowheight + y_top_pad + ref_boxheight / 2
+
+        boxheight = ref_boxheight
+        boxwidth = max(ref_boxwidth, _get_text_width(v.text))
+        boxx = x - boxwidth / 2
+        boxy = y - boxheight / 2
+
+        if xi == 0:
+            # Qubits
+            t += _rect(boxx, boxy, boxwidth, boxheight, strokewidth=0)
+            t += _text(x, y, v.text)
+            continue
+
+        if v.text == '@':
+            t += f'<circle cx="{x}" cy="{y}" r="{ref_boxheight / 4}" />'
+            continue
+        if v.text == '×':
+            t += _text(x, y + 3, '×', fontsize=40)
+            continue
+
+        t += _rect(boxx, boxy, boxwidth, boxheight)
+        t += _text(x, y, v.text, fontsize=14 if len(v.text) > 1 else 18)
+
+    t += '</svg>'
+    return t
+
+
+class SVGCircuit:
+
+    def __init__(self, circuit: 'cirq.Circuit'):
+        # coverage: ignore
+        self.circuit = circuit
+
+    def _repr_svg_(self) -> str:
+        # coverage: ignore
+        tdd = self.circuit.to_text_diagram_drawer(transpose=False)
+        return tdd_to_svg(tdd)
+
+
+def svg_circuit(circuit: 'cirq.Circuit') -> str:
```

Comment:
This doesn't return an `SvgCircuit`, which is a bit confusing. Maybe `circuit_to_svg`?

Reply:
done